File: apngdrawable/src/main/java/com/dragon/drawable/apng/FrameData.java
package com.dragon.drawable.apng;

import android.graphics.Bitmap;

/**
 * @author chenjiulong
 */
public class FrameData {
    public FrameData(Bitmap bitmap) {
        this.bitmap = bitmap;
    }

    public Bitmap bitmap;
    public int frameCount;
    public int index = 0;
    public long delay = 0;
    public long firstDrawTime = 0;
}

File: APngLib/src/main/java/com/dragon/apnglib/FrameData.kt
package com.dragon.apnglib

import android.graphics.Bitmap

/**
 * @author dragon
 */
internal class FrameData(val bitmap: Bitmap) {
    var frameCount = 0
    var index = 0
    var delay = 0L
    fun release() {
        if (!bitmap.isRecycled) {
            bitmap.recycle()
        }
    }

    fun reset() {
        frameCount = 0
        index = 0
        delay = 0
    }
}